### PR TITLE
refactor(BasicSwitch): to composition api

### PR DESCRIPTION
**Pull request by leo-anderson-x**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring

## Context

- [x] Related wtih #4750 
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [x] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=12RA5J7wGg6bDWPTZG359Pn5bCzXSJRfsifo3UktqVAjhJ9k&usdamount=0&donation=true)

#### Community participation

- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

### <samp>🤖 Generated by Copilot at 961370c</samp>

Refactored BasicSwitch component to use `<script setup>` syntax for better readability and maintainability.

### <samp>🤖 Generated by Copilot at 961370c</samp>

> _`<script setup>` shines_
> _Props, emits, and variables_
> _Clear as autumn sky_

### CODE REVIEW

**Comment by yangwao** [comment]
are you on latest branch?

**Comment by leo-anderson-x** [comment]
@yangwao 

> are you on latest branch?

Yes. But it says "out-of-date with the base branch" soon after the PR is submitted. I just merged main, now it's up to date.

File: components/shared/form/BasicSwitch.vue [hunk 1]
```diff
@@ -15,29 +15,37 @@
...
+  set: (value: boolean) => {
+    emit('input', value)
+  },
+})
+
+const componentName = computed(() => {
+  return props.message ? NeoTooltip : 'span'
+})
+
+const properLabel: TranslateResult = computed(() => {
```
Comment (stephenjason89) [inline components/shared/form/BasicSwitch.vue:47]: type should be ComputedRef<TranslateResult>

**Comment by stephenjason89** [comment]
Other than https://github.com/kodadot/nft-gallery/pull/6814/files#r1299373805
code lgtm 😆

File: components/shared/form/BasicSwitch.vue [hunk 2]
```diff
@@ -15,29 +15,33 @@
   </NeoField>
 </template>
 
-<script lang="ts">
-import { TranslateResult } from 'vue-i18n'
-import { Component, Prop, VModel, Vue } from 'nuxt-property-decorator'
+<script lang="ts" setup>
+import type { ComputedRef } from 'vue'
```
Comment (roiLeo) [inline components/shared/form/BasicSwitch.vue:19]:
```suggestion
```

File: components/shared/form/BasicSwitch.vue [hunk 3]
```diff
@@ -15,29 +15,33 @@
...
-    return this.$t(this.isSwitched ? this.label : offLabel)
-  }
-}
+const isSwitched = useVModel(props, 'value', emit, { eventName: 'input' })
+
+const componentName = computed(() => {
+  return props.message ? NeoTooltip : 'span'
+})
+
+const properLabel: ComputedRef<TranslateResult> = computed(() => {
```
Comment (roiLeo) [inline components/shared/form/BasicSwitch.vue:43]:
```suggestion
const properLabel = computed<TranslateResult>(() => {
```

**Review by roiLeo — APPROVED** [review]

**Comment by prury** [comment]
nft/collection creation switchs ✔️ 
user offers ✔️ 
simplemint ✔️

**Review by vikiival — APPROVED** [review]

**Comment by vikiival** [comment]
pay 20 usd

**Pull request merged**